docs: Restructure Deploy your app section with Custom hosting (#539)
serverpod/serverpod_docs: docs/08-deployments/custom-hosting/01-choosing-a-strategy.md

---
sidebar_label: Choosing a strategy
description: Choose how to self-host a Serverpod server, on a server cluster, a serverless platform, or your own machine, as an alternative to Serverpod Cloud.
---

# Choosing a strategy

> You can run a Serverpod server anywhere it runs as a Dart process or a Docker container: a managed server cluster, a serverless platform, or your own machine. This page covers the trade-offs, and the following pages walk through each path. For the managed path, see [Serverpod Cloud](/cloud).

Reviewer: ```suggestion
You can run a Serverpod server anywhere that supports a Dart process or a Docker container: a managed server cluster, a serverless platform, or your own machine. This page covers the trade-offs, and the following pages walk through each path. For the managed path, see [Serverpod Cloud](/cloud).
```

|      | Server cluster                                                                            | Serverless                                                          |
| :--- | :---------------------------------------------------------------------------------------- | :------------------------------------------------------------------ |
| Pros | All features are supported. Great for real-time communication. Cost-efficient at scale.   | Minimal starting cost. Easier configuration. Minimal maintenance.   |
| Cons | Slightly higher starting cost. More complex to set up.                                    | Limited feature set. The server cannot hold state.                  |

Serverless does not support:

- Future calls. (Possible to configure, but requires a more advanced setup.)
- Health metrics.
- On-server caching. Caching can still happen while a serverless instance stays warm, but it can be lost at any time. Caching with Redis is supported.
- In-memory state. Store shared state in an external service such as Postgres, Redis, or another API instead.

> ## Pick a guide

Reviewer: "Pick a guide" branches off, and then come the approximate costs.
Consider whether approximate costs belong to the serverless vs server cluster comparison above.

Author: This section is pre-existing content from 01-deployment-strategy.md — it only shows as new because the file was renamed. Reordering it is out of scope for a restructure PR
